trio/_tests/check_type_completeness.py

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env python3
+# this file is not run as part of the tests, instead it's run standalone from check.sh
+import subprocess
+import json
+from pathlib import Path
+from typing import Any
+import sys
+import argparse
+
+RES_FILE = Path(__file__).parent / "verify_types.json"
+failed = False
+
+
+def run_pyright():
+    return subprocess.run(
+        ["pyright", "--verifytypes=trio", "--outputjson", "--ignoreexternal"],
```

Comment:
This looks cool, but why `--ignoreexternal`?

Reply:
> If the --verifytypes option is combined with --ignoreexternal, any incomplete types that are imported from other external packages are ignored. This allows library authors to focus on adding type annotations for the code that is directly under their control.

I'll add a comment (I didn't remember either :innocent:)

Reply:
I'm not fully sure if it makes a difference, but could try removing it later on and potentially open PR's to any external libraries that affects us.